### ISSUE TITLE
Home Assistant is not updating

### ISSUE THREAD

**Issue by eeezyy** [ronschaeffer/sbm2mqtt#13]
I am having issues connecting Home Assistant to MQTT.

Here's my `configuration.yaml`
```
mqtt:
  client_id: home_assistant
  broker: 193.168.178.39
  port: 8123
  username: admin
  password: password
  discovery: true
  birth_message:
    topic: 'home-assistant/status'
    payload: 'online'
  will_message:
    topic: 'home-assistant/status'
    payload: 'offline'

sensor:
- platform: mqtt
  name: 'switchbot_meter_temperature'
  state_topic: 'switchbot_meter/<the-mac-address-from-output>' # MAC address of this meter
  value_template: '{{ value_json.temperature }}'
  unit_of_measurement: '°C' # Change to '°F' as appropriate
- platform: mqtt
  name: 'switchbot_meter_humidity'
  state_topic: 'switchbot_meter/<the-mac-address-from-output>' # MAC address of this meter
  value_template: '{{ value_json.humidity }}'
  unit_of_measurement: '%'
  icon: mdi:water-percent
- platform: mqtt
  name: 'switchbot_meter_battery'
  state_topic: 'switchbot_meter/<the-mac-address-from-output>' # MAC address of this meter
  value_template: '{{ value_json.battery }}'
  unit_of_measurement: '%'
  icon: mdi:battery
```

I see that Home assistant is connected to the MQTT Broker, to be sure I set the client_id.

I also have the MQTT Integration Plugin installed (I think this is needed).

On the card in the dashboard the values stay at "Unknown".

What am I doing wrong? Is there a step I'm missing?
Don't have experience using MQTT yet (using HiveMQ Broker at the moment).

**Comment by ronschaeffer** [comment]
It looks like your port number is probably wrong in the config file. 8123
is the port for Home Assistant itself. The MQTT port is probably 1883.

Be sure also that the username and password in the config file are the same as whichever you set in the MQTT add on configuration. Don't use your Home Assistant user name and password there.

Also, check the log of the MQTT add on to be sure that it is up and running
without errors.

Finally, you say that you're using HiveMQTT. That shouldn't be necessary. Just use the standard Mosquitto add on.

**Comment by eeezyy** [comment]
Thank you for your reply.

The port was indeed wrong. I have tested it before without that line with the default port (1883), which also did not work.
I also see no errors in the logs. When the server was down there were some error logs to see.
So I think it works in general.

One other mistake was to configure the MQTT add on over UI and also configuring it in the configuration.yaml.
I see that the add on will show up just after adding the configuration.yaml parts.

I don't have the supervisor installed within home assistant so I don't see the Mosquitto add on.

**Comment by eeezyy** [comment]
I tried to pin it down using a MQTT client.
I've send the same JSON to the same topic and it worked.

After that I tried the script again and saw that home assistant is reacting to it (shows value as unknown) and then shows the values from before.
Something seems to be broken there.

```
mqtt_host = "193.168.178.39"
mqtt_port = 1883
mqtt_timeout = 30
mqtt_client = "switchbot_meter_checker"
mqtt_user = "admin"
mqtt_pass = "password"
mqtt_topic = "smartbot_meter/"
``` 

But I don't see what could be the cause the problem.

Do you have an advice how to debug the request of the script?

I also can't see anything that could cause the problem in the source...
`mqttc.publish(mqtt_topic + mac, msg_data, 1)`

**Comment by eeezyy** [comment]
I've updated the paho-mqtt python library and all other packages.
Also changed to eclipse-mosquitto as a separate docker installation without any progress.

I also don't see any connection in the UI.
Is there a difference which mqtt_client is used?

The observed reaction from home assistance I described before is only happening when using the same client used in Home Assistant itself...

Maybe I will wait for the docker instance to exclude any side effects on my system.

**Comment by ronschaeffer** [comment]
@eeezyy wrote:
`mqtt_topic = "smartbot_meter/"`

Did you change the topic of your HA sensors to be "smartbot_meter/" as you have in your sbm2mqtt_config.py file? If not, that is the problem--or at least one of them.

Use `mqtt_topic = "switchbot_meter/"` in the sbm2mqtt_config.py file, ensure that the topic of your HA sensors is the same, and see if that works.

**Comment by eeezyy** [comment]
Good catch. 
I should not change too much in the config ;)

But this is not the problem.
After fixing it, it behaves as before...

Used the output also in MQTT client and it works there.

**Comment by eeezyy** [comment]
Is the docker implementation of the PR working?
Then I would try it :)

**Comment by eeezyy** [comment]
Okay, I got it working now.
Don't know what was the actual problem, but I think I did changed to much and tested different settings.

Somehow when MQTT is not able to connect because of wrong credentials don't show any errors.
Maybe this can be improved.

Thanks for your time and your help and of course for this project!

Now that this is working, I will have a look into the switchbot bot, to turn some switches off and on :)
